code/datums/components/laser_components.dm

```diff
@@ -0,0 +1,32 @@
+/datum/component/extralasers
+	dupe_mode = COMPONENT_DUPE_UNIQUE_PASSARGS
+	var/path
+
+/datum/component/extralasers/Initialize(ammo, _lens, _path)
+	if(!istype(parent, /obj/item/gun/energy/laser))
+		return COMPONENT_INCOMPATIBLE
+	var/obj/item/gun/energy/laser/L = parent
+	L.ammo_type  += new ammo (src)
+	path = _path
+	qdel(_lens)
+
+/datum/component/extralasers/RegisterWithParent()
+	RegisterSignal(parent, COMSIG_PARENT_ATTACKBY, .proc/attackby)
+
+/datum/component/extralasers/UnregisterFromParent()
+	UnregisterSignal(parent, COMSIG_PARENT_ATTACKBY)
+
+/datum/component/extralasers/proc/attackby(datum/source, obj/item/I, mob/user, params)
+	if(I.tool_behaviour == TOOL_CROWBAR)
+		var/obj/item/gun/energy/laser/L = parent
+		L?.chambered = null
+		if(L.ammo_type)
+			var/obj/item/ammo_casing/energy/R = LAZYACCESS(L.ammo_type, L.ammo_type.len)
```

Comment:
How do you know your ammo type is at the end? You should be keeping track of what you added to remove it properly